### PR TITLE
Fix/core adjustments

### PR DESCRIPTION
**Pull request by integricho**

### CODE REVIEW

File: librarian/commands/facets.py [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
-def refill_facets(arg, supervisor):
-    print('Begin facets refill.')
-    config = supervisor.config
-    archive = FacetsArchive(supervisor.exts.fsal,
-                            supervisor.exts.databases.facets,
-                            config=config)
-    archive.clear_and_reload()
-    print('Facet refill finished.')
-    raise supervisor.EarlyExit()
+class RefillFacetsCommand(object):
```
Comment (integricho) [inline librarian/commands/facets.py]: we shall prepare a base command class to inherit from

File: librarian/hooks.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    for command_path in exts.config['command.handlers']:
+        command_cls = import_attr(command_path)
+        exts.commands.register(command_cls.name,
+                               command_cls(),
+                               command_cls.option,
+                               command_cls.action,
+                               command_cls.help)
+
+
+def register_menu_items():
```
Comment (integricho) [inline librarian/hooks.py]: if we unify the apis of the registration methods, that is all can just accept a class without extra calls, we can generalize these register_xx_ functions

File: librarian/setup/auth.py [hunk 3]
```diff
@@ -1,34 +1,45 @@
...
+from ..core.contrib.auth.utils import generate_random_key
+from ..data.setup.decorators import autoconfigure
+from ..forms.auth import RegistrationForm
+
+
+autoconfigure('session.secret')(generate_random_key)
+autoconfigure('csrf.secret')(generate_random_key)
+
+
+class SuperuserStep:
```
Comment (integricho) [inline librarian/setup/auth.py]: Class based route handlers should be considered here

Comment (foxbunny) [inline librarian/setup/auth.py]: Wizard code should provide a base class that let's you write steps in a similar way as how it's done in `streamline.FormRoute`.

**Comment by foxbunny** [comment]
```
Component librarian encountered an error during it's loading, installation skipped.
Traceback (most recent call last):
  File "/home/hajime/code/librarian/librarian/core/dependencies.py", line 64, in _import_modules
    mod = importlib.import_module(mod_path)
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/hajime/code/librarian/librarian/hooks.py", line 8, in <module>
    from .data.setup import Setup, SetupWizard
  File "/home/hajime/code/librarian/librarian/data/setup.py", line 19, in <module>
    from .decorators import AUTO_CONFIGURATORS
ImportError: No module named decorators
Traceback (most recent call last):
  File "/home/hajime/.virtualenvs/libtest/bin/librarian", line 9, in <module>
    load_entry_point('librarian', 'console_scripts', 'librarian')()
  File "/home/hajime/code/librarian/librarian/app.py", line 29, in main
    supervisor = Supervisor(root_dir)
  File "/home/hajime/code/librarian/librarian/core/supervisor.py", line 76, in __init__
    self._load_components()
  File "/home/hajime/code/librarian/librarian/core/supervisor.py", line 184, in _load_components
    for member in loader.load():
  File "/home/hajime/code/librarian/librarian/core/dependencies.py", line 178, in load
    self._parse()
  File "/home/hajime/code/librarian/librarian/core/dependencies.py", line 138, in _parse
    raise UnresolvableDependency(msg)
librarian.core.dependencies.UnresolvableDependency: Dependency 'librarian_core.contrib.system.routes.routes' is missing.
```

Any ideas?

**Comment by foxbunny** [comment]
Oh, <FACEPALM>, I'm using an outdated and unadjusted config from the previous releases.

**Comment by goodbadwolf** [comment]
The import path is incorrect though. It should be:
`from ..decorators.setup import AUTO_CONFIGURATORS`

**Comment by foxbunny** [comment]
Right.

**Comment by foxbunny** [comment]
@integricho Can we have a reference config that works with the restructured Librarian?

**Comment by integricho** [comment]
Just delete all the entries from under `+components`, otherwise projectwise there still are millions of importerrors

**Comment by foxbunny** [comment]
Need any help dealing with those?

**Comment by integricho** [comment]
Sure :) I haven't even yet started on it, due to this being still unmerged into chore/monolithification

**Pull request merged**